server/config.go

```diff
@@ -496,6 +512,12 @@ func CLIFlags() []cli.Flag {
 			Value:   cli.NewStringSlice(),
 			EnvVars: prefixEnvVars("CACHE_TARGETS"),
 		},
+		&cli.BoolFlag{
+			Name:    ReadOnlyFlagName,
+			Usage:   "Whether to run the proxy in read-only mode; set to true to only allow retrieval from EigenDA.",
+			Value:   false,
+			EnvVars: prefixEnvVars("READ_ONLY"),
+		},
```

Comment:
let's also update the readme documentation so users know how to target this functionality